api/turing/api/deployment_controller.go

```diff
@@ -190,41 +191,20 @@ func (c RouterDeploymentController) deployRouterVersion(
 		}
 	}
 
-	expSvc := c.BaseController.AppContext.ExperimentsService
 	if routerVersion.ExperimentEngine.Type != models.ExperimentEngineTypeNop {
-		experimentConfig = routerVersion.ExperimentEngine.Config
-		isClientSelectionEnabled, err := expSvc.IsClientSelectionEnabled(routerVersion.ExperimentEngine.Type)
+		experimentConfig, err = c.getExperimentConfig(routerVersion)
 		if err != nil {
 			return "", c.updateRouterVersionStatusToFailed(err, routerVersion)
 		}
-		if isClientSelectionEnabled {
-			// Convert the config to the standard type
-			standardExperimentConfig, err := manager.ParseStandardExperimentConfig(experimentConfig)
+
+		if routerVersion.ExperimentEngine.PluginConfig.ServiceAccount != nil {
```

Comment:
I think we should also check for `PluginConfig` != nil ?

On a related note, I wonder if the `ServiceAccount` should be promoted up (directly under [ExperimentEngine](https://github.com/caraml-dev/turing/blob/main/api/turing/models/experiment_engine.go#L17)), as it's not limited to RPC plugins (theoretically it can also be used for the in_proc experiment engines, right?). Along the same lines, when we are taking the configs in for the experiment engines, we could take it in as a separate config, at the same level as `ExperimentEnginePluginConfig`.

 So, the helm values would look like: https://github.com/caraml-dev/turing/blob/main/infra/charts/turing/values.yaml#L107

(Example considers this comment as well: https://github.com/caraml-dev/turing/pull/274/files#r1013555923)

```
Example:
  - name: my-exp-engine
    type: rpc-plugin
    serviceAccountKeyFilePath: /etc/abc.json
    rpcPlugin:
      image: ghcr.io/turing/my-exp-engine:latest
    options:
      key-1: value-1
```

And may be, in `config.go`, we need to change to:
```
ExperimentEnginePlugins map[string]*ExperimentEngineConfig `validate:"dive"`
```

where `ExperimentEngineConfig` is:
```
type ExperimentEngineConfig {
    PluginConfig *ExperimentEnginePluginConfig
    ServiceAccountKeyFilePath *string
}
```